### DISCUSSION TITLE
Request: Allow Controlling Clickability of Headings in README Files

### DISCUSSION THREAD

**Discussion by danielbrodi** [community/community#68054] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I've noticed that headings in README files on GitHub are automatically generating clickable links to anchor points within the README. While this feature can be useful in some cases, it can also lead to unintended behavior when the intention is to have non-clickable headings.

## Issue:

**Automatic Clickability:** Currently, all headings in README files, whether written as `<h1>`, `<h2>`, or using Markdown's `#`, are automatically generating clickable links. This behavior seems to have changed recently, and it results in headings behaving like links by default.
For example: `### Technologies` will automatically create an anchor point to: `https://github.com/daniel#technologies`.

**Intention for Non-Clickable Headings:** In some cases, README authors may prefer to have headings that are not clickable, especially when the intention is not to link to sections within the README. This behavior may be more in line with the intended use of headings for organizational purposes.

### I would like to request an enhancement to GitHub's Markdown rendering behavior to allow authors more control over the clickability of headings in README files:

**Optional Clickability:** Provide an option or syntax that allows authors to specify whether a heading should be clickable or not. For example, using an attribute or flag in the Markdown syntax.

**Consistency:** Ensure that the default behavior remains consistent with the intended use of headings for organization and structure, which is typically non-clickable.

## Examples of possible solution:

```
<!-- A potential syntax for specifying non-clickable headings -->
<h1 clickable="false">Welcome to Daniel's GitHub</h1>
```

`[clickable=false]### Technologies`

**Comment by kamlendras** [comment]


**Reply by danielbrodi** [reply]
Thank you for the quick response. I wasn't entirely sure whether to categorize this as feedback, a feature request, or even a bug report because I'm uncertain if the behavior with anchors was intentional. 

Can you please clarify if this behavior with anchors was intentionally added by GitHub in the last update?

**Comment by kamlendras** [comment]


**Reply by danielbrodi** [reply]
To be honest, it wasn't like that until 2 or 3 weeks ago, and this recent change led me to raise the issue.
Also, it's not just me, many others report that their profile suddenly appears to be all 'blueish' and full of clickable headers after years they keep the same README.

**Reply by mspronesti** [reply]
It wasn't like this until last week. I check my profile on a daily basis and I'm sure this change took effect on Wednesday. 
As a result, I renamed my GitHub profile README. It become rather ugly and I don't want it anymore on my landing page.
Hope you can fix this asap.
Lately you guys are putting a lot of effort into fixing things that are not broken.

**Reply by ramblingadam** [reply]
This is such a bizarre, unnecessary, and unasked-for change. I can't imagine why Headings should _ever_ be links, let alone all of them!

My carefully designed project ReadMe's look ridiculous with all the headings converted into blue links that don't even go anywhere.

**Comment by nelsonaloysio** [comment]
As a workaround, I have resorted to inserting a zero-width character with a hyperlink in the beginning of any headings:

```
<h1><a href="#">&#x200B;</a>Heading</h1>
```

The same applies to Markdown:

```
# [&#x200B;](#) Heading
```

In both cases, the heading is displayed as pure text.

**Reply by danielbrodi** [reply]
What a simple yet clever solution, thank you for sharing with us!
I'm not marking your comment as the answer, because I still want GitHub to fix this glitch without the need to find any workarounds.

**Reply by MattHammond94** [reply]
Big shout out to @nelsonaloysio!

**Reply by Timmichi** [reply]
@danielbrodi It looks like GitHub fixed this.

**Reply by kamlendras** [reply]
Not properly :(
